ext/reflection/php_reflection.c

```diff
@@ -2261,6 +2254,12 @@ ZEND_METHOD(ReflectionParameter, __construct)
 			_DO_THROW("The parameter specified by its name could not be found");
 			goto failure;
 		}
+	} else {
+		position= (int)int_parameter;
```

Comment:
I'd leave only the zend_long variable. We're casting below anyway.

Reply:
The `position` variable is referenced below therefore I don't think we can just get rid of it like that.
```
ref->arg_info = &arg_info[position];
ref->offset = (uint32_t)position;
ref->required = (uint32_t)position < fptr->common.required_num_args;
```

Reply:
What I mean here is that you can just use `zend_long position` and accept `position` in ZPP. I don't think there is a need for two variables, it will only introduce subtle bugs (e.g. here you are truncating the value *before* the comparison directly below).

Reply:
Okay, finally understood what you meant.
Merged both of them together, although I don't know if the name I chose is ideal...
Also added a ValueError when the position provided is negative as this should throw early IMHO